mmengine/optim/optimizer/optimizer_wrapper.py

```diff
@@ -276,7 +301,16 @@ def param_groups(self) -> List[dict]:
         Returns:
              dict: the ``param_groups`` of :attr:`optimizer`.
         """
-        return self.optimizer.param_groups
+
+        # If ``self.base_param_settings`` is not None, the ``param_groups``
+        # of :attr:`optimizer` will be appended to
+        # ``self.base_param_settings``. Otherwise, the ``param_groups``
+        # of :attr:`optimizer` will be returned.
+        if self.base_param_settings is not None:
+            return list(
+                chain(self.optimizer.param_groups, [self.base_param_settings]))
+        else:
+            return list(self.optimizer.param_groups)
```

Comment:
`self.optimizer.param_groups` is already a list, so there is no need to use a list for conversion here.

Reply:
yep.

Reply:
I discovered that the failure occurred because I changed the unit test. I have reverted those modifications and restored the unit test to its original state. As a result, we no longer require the list that contains `self.optimizer.param_groups`.

Reply:
https://github.com/open-mmlab/mmengine/pull/1019/files#diff-765cf7329b9aac3e0abe0e297b5cb099afd539a3500628ec5cbf5ee2ebbe0432R517